libs/ngxtension/trackby-id-prop/src/trackby-id-prop.ts

```diff
@@ -0,0 +1,42 @@
+//INSPIRED BY https://medium.com/ngconf/make-trackby-easy-to-use-a3dd5f1f733b
+import { NgForOf } from '@angular/common';
+import {
+	Directive,
+	Input,
+	Provider,
+	inject,
+	type NgIterable,
+} from '@angular/core';
+
+@Directive({
+	selector: '[ngForTrackById]',
+	standalone: true,
+})
+export class NgForTrackByIdDirective<T extends { id: string | number }> {
+	@Input() ngForOf!: NgIterable<T>;
+	private ngFor = inject(NgForOf<T>, { self: true });
+
+	constructor() {
+		this.ngFor.ngForTrackBy = (index: number, item: T) => item.id;
+	}
+}
+
+@Directive({
+	selector: '[ngForTrackByProp]',
+	standalone: true,
+})
+export class NgForTrackByPropDirective<T> {
+	@Input() ngForOf!: NgIterable<T>;
+	private ngFor = inject(NgForOf<T>, { self: true });
+
+	@Input()
+	set ngForTrackByProp(trackByProp: keyof T) {
+		if (!trackByProp) return; //throw new Error("You must specify trackByProp:'VALID_PROP_NAME'");
```

Comment:
comment: maybe with `{required: true}`, we don't need to check here or throw error instead of early return

Reply:
I have only one concern (before {required}) and was that the user mis-use the directive writing only `trackByProp` forgetting to specify the value, so the setter will be called but with '' o undefined I don't know, so I decided to NOT throw but simply do nothing, at the end ngFor works without trackBy with a default behavior of tracking index, so if the setter is called with '', null, undefined I simply "opt-out" to trackBy and fallback to the default ngFor behavior without erroring.

I don't ever test what happen if {required} is set for Input, but the user specify and explicit value of `trackByProp:''` so if you don't mind I prefer to keep my _"defense programming"_ and check the value before using.